pkg/agent/agent.go

```diff
@@ -343,17 +343,31 @@ func (a *Agent) probeServerInfo() error {
 	return err
 }
 
+var ErrNoIdentityHostname = errors.New("the device doesn't have a valid hostname or identity. Set PREFERRED_IDENTITY or PREFERRED_HOSTNAME to specify the device's name and identity")
```

Comment:
This error is not only related to the hostname, but to the identity as well. So the variable name should refer to that.

Reply:
Something like `ErrNoIdentityAndHostname` ?

Reply:
I believe so. Also, you should adjust the comment inside of the code to mention that both information are missing.